libraries/botbuilder-dialogs/src/memory/dialogStateManager.ts

```diff
@@ -79,70 +80,91 @@ export class DialogStateManager {
      * @param defaultValue (Optional) default value to use if the path isn't found. May be a function that returns the default value to use.
      * @returns The found value or undefined if not found and no `defaultValue` specified.
      */
-    public getValue<T = any>(pathExpression: string, defaultValue?: T|(() => T)): T {
-        function returnDefault(): T {
-            return typeof defaultValue == 'function' ? (defaultValue as Function)() : defaultValue;  
+    public getValue<T = any>(pathExpression: string, defaultValue?: T | (() => { value: T, error: string })): { value: T, error: string } {
```

Comment:
>{ value: T, error: string } [](start = 110, length = 27)

Why are we making this change?  Now every caller should technically be looking for an error and re-throwing. It's way simpler to just have the state manager throw an error.  I looked at the C# code and they're not doing this.

Also, there shouldn't be any reason to eat errors as the state manager already returns "undefined" for an missing path.  What's being solved here?

Reply:
In this PR: #1466  they introduced MemoryInterface in botframework-expression, so we have to let DialogStateManager implement MemoryInterface to evaluate expressions properly.
However, the GetValue method in MemoryInterface differs from DialogStateManager’s, so I changed the api to fit MemoryInterface’s.
Is there a better solution for us to keep the old api of DialogStateManager so we don’t need to make this change?

Reply:
Reverted changes to keep the original DSM api.